src/crimson/net/Dispatcher.h

```diff
@@ -16,43 +16,114 @@
 
 #include <seastar/core/future.hh>
 #include <seastar/core/sharded.hh>
+#include "crimson/common/gated.h"
 
 #include "Fwd.h"
 
 class AuthAuthorizer;
 
 namespace crimson::net {
 
-class Dispatcher {
+class Dispatcher : public crimson::common::Gated {
```

Comment:
I'm not very convinced to simply add another layer of gate (per-dispatcher) onto the existing per-connection gate `crimson::net::Protocol::gate` which serves the same purpose. This is on the critical path and thus we need to be careful about any extra overhead.

If we are making the decision to not gate event dispatching by the connection gate, IMHO we need to make those methods to return `void` to connections. The according futures need to be gated by the dependent components themselves (maybe even not in the `Dispatcher`).

Reply:
@cyx1231st Hi, thanks for the review, I agree with you. After a second thought, I think those gates in components other than the messenger serves only one purpose ---- preventing further processing of events ---- which can be achieved by simply adding a boolean flag "stopping" just like that in SocketMessenger. What do you think?:-)

If we have an agreement about this, I can get down modifying the code:-)

Reply:
> ---- preventing further processing of events ---- which can be achieved by simply adding a boolean flag "stopping" instead, just like that in SocketMessenger. 

Looks reasonable to me, this seems to be simpler.